src/schema.ts

```diff
@@ -261,31 +272,65 @@ UserSchema.virtual("oldEnoughForWithdrawal").get(function (this: typeof UserSche
   return d - this.created_at.getTime() > yamlConfig.limits.oldEnoughForWithdrawal
 })
 
-UserSchema.methods.limitHit = async function ({
-  on_us,
-  amount,
-}: {
-  on_us: boolean
-  amount: number
-}) {
-  const timestampYesterday = Date.now() - MS_PER_DAY
+UserSchema.virtual("twoFactorEnabled").get(function (this: typeof UserSchema) {
+  return !!this.twoFactor.secret
+})
 
-  const txnType = on_us
-    ? [{ type: "on_us" }, { type: "onchain_on_us" }]
-    : [{ type: { $ne: "on_us" } }]
+const getTimestampYesterday = () => Date.now() - MS_PER_DAY
 
-  const limit = yamlConfig.limits[on_us ? "onUs" : "withdrawal"].level[this.level]
+// FIXME: Improve function name
+UserSchema.methods.remainingTwoFactorLimit = async function () {
+  const threshold = this.twoFactor.threshold
+
+  const txnType = [
+    { type: "on_us" },
```

Comment:
@krtk6160 we need to differentiate on_us to on_us_receipt and on on_us_sent. otherwise the velocity check would also apply for money received.

we could open a ticket and do this in another PR 

cc @samerbuna for the graphql type

Reply:
https://github.com/GaloyMoney/galoy/issues/320